pipeline/common/downloads.py

```diff
@@ -18,19 +18,28 @@
 logger = get_logger(__file__)
 
 
+class DownloadException(Exception):
```

Comment:
Ah, this is a nice change here.

Reply:
Yeah, it's a good practice to throw specific exceptions so that the `except` code doesn't catch everything. Python has very few built-in ones. Usually, when I don't want to create a custom one, I throw `ValueError`, which is a generic user code exception. In this case, having a custom type makes sense.